iac_controller.py

```diff
@@ -0,0 +1,36 @@
+import os
+from collections import deque
+from typing import Collection
+
+import requests
+from requests.auth import HTTPBasicAuth
+
+
+class IACController:
+
+    def __init__(self, max_total_iacs: int, batch_size: int = 1000):
+        self._iac_url = os.getenv('IAC_URL')
+        self._iac_auth = HTTPBasicAuth(os.getenv('IAC_USERNAME'), os.getenv('IAC_PASSWORD'))
+        self._iac_pool = deque()
+        self._max_total_iacs = max_total_iacs
+        self._batch_size = batch_size
+        self._total_iacs_fetched = 0
+
+    def get_iac(self) -> str:
+        if not self._iac_pool:
+            self._add_iac_batch_to_pool()
+        return self._iac_pool.pop()
+
+    def _add_iac_batch_to_pool(self):
```

Comment:
could this method be async so that it wouldn't cause delays when it's topping up the pool?

Reply:
I'll have a look at doing that but I think the time spent putting the codes on the queue is negligible compared to the time spent in the HTTP call.

Reply:
For what it's worth I would keep it simple for now.  There's not a problem with speed at the moment.

Reply:
I think I misunderstood you in my first reply, but I agree with Luke. If we find we need this script to be faster in the future, that's probably the first thing to try.